### PR TITLE
ENH: add figure id to its __repr__

### PR DESCRIPTION
**Pull request by jklymak**
## PR Summary

Per #13344 I don't see any reason figure.__repr__ should *not* include the object id in it...
`print(fig.__repr__())` now returns:

```
<Figure size 1280x960 with 1 Axes; 0x109147cf8>
```

## PR Checklist

- [ ] Has Pytest style unit tests
- [ ] Code is [Flake 8](http://flake8.pycqa.org/en/latest/) compliant
- [ ] New features are documented, with examples if plot related
- [ ] Documentation is sphinx and numpydoc compliant
- [ ] Added an entry to doc/users/next_whats_new/ if major new feature (follow instructions in README.rst there)
- [ ] Documented in doc/api/api_changes.rst if API changed in a backward-incompatible way

### CODE REVIEW

File: lib/matplotlib/figure.py [hunk 1]
```diff
@@ -266,11 +266,9 @@ def __str__(self):
...
 
     def __repr__(self):
-        return "<{clsname} size {h:g}x{w:g} with {naxes} Axes>".format(
-            clsname=self.__class__.__name__,
-            h=self.bbox.size[0], w=self.bbox.size[1],
-            naxes=len(self.axes),
-        )
+        return (f"<{self.__class__.__name__} size {self.bbox.size[0]:g}"
+                f"x{self.bbox.size[1]:g} with {len(self.axes)} Axes; "
+                f"0x{id(self):x}>")
```
Comment (timhoffm) [inline lib/matplotlib/figure.py]: Default reprs read like `<__main__.Foo instance at 0x2afd73c49518>`. Maybe we should also use `at 0x...` to be more consistent the the standard repr.

File: lib/matplotlib/tests/test_figure.py [hunk 2]
```diff
@@ -373,7 +373,7 @@ def test_savefig():
 
 def test_figure_repr():
     fig = plt.figure(figsize=(10, 20), dpi=10)
-    assert repr(fig) == "<Figure size 100x200 with 0 Axes>"
+    assert repr(fig)[:36] == "<Figure size 100x200 with 0 Axes; 0x"
```
Comment (timhoffm) [inline lib/matplotlib/tests/test_figure.py]: Would be slightly better with `assert re.match()`.

**Comment by QuLogic** [comment]
I think removing the object address was the point of #9372, wasn't it?

**Comment by jklymak** [comment]
I don’t see why you’d need to remove the address I just assumed it was to make the repr more readable.

**Comment by NelleV** [comment]
ping @Carreau

**Comment by jklymak** [comment]
Ooops, ha ha, I didn't read to the top of #9372.  We have one library that *doesn't* want unique ids in the `__repr__` and one that does.  I guess given the existence of `id(object)` I'll close this...

**Pull request closed without merging**